examples/tutorials/api/observation_clustering.py

```diff
@@ -0,0 +1,220 @@
+
+- Grouping the observations depending on the IRF quality, by means of hierarchical clustering.
+
+"""
+
+
+import numpy as np
+import astropy.units as u
+from astropy.coordinates import SkyCoord
+import matplotlib.pyplot as plt
+from gammapy.data import DataStore
+from gammapy.data.observations import Observations
+from gammapy.data.utils import get_irfs_features
+from gammapy.utils.cluster import hierarchical_clustering, standard_scaler
+
+######################################################################
+# Obtain the observations
+# -----------------------
+#
+# First need to define the `~gammapy.data.DataStore` object for the HESS DL3 DR1
+# data. Next, utilise a cone search to select only the observations of interest.
+# In this case, we choose PKS 2155-304 as the object of interest.
+#
+# The `~gammapy.data.ObservationTable` is then filtered using the `select_observations` tool.
+#
+
+data_store = DataStore.from_dir("$GAMMAPY_DATA/hess-dl3-dr1")
+
+selection = dict(
+    type="sky_circle",
+    frame="icrs",
+    lon="329.71693826 deg",
+    lat="-30.2255 deg",
+    radius="2 deg",
+)
+obs_table = data_store.obs_table
+selected_obs_table = obs_table.select_observations(selection)
+
+
+######################################################################
+# More complex selection can be done by utilising the obs_table entries directly.
+# We can now retrieve the relevant observations by passing their obs_id to the
+# `~gammapy.data.DataStore.get_observations` method.
+#
+
+obs_ids = selected_obs_table["OBS_ID"]
+observations = data_store.get_observations(obs_ids)
+
+
+######################################################################
+# Show various observation quantities
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+# Print here the range of zenith angles and muon efficiencies, to see
+# if there is a sensible way to group the observations.
+#
+
+obs_zenith = selected_obs_table["ZEN_PNT"]
+obs_muoneff = selected_obs_table["MUONEFF"]
+
+print(f"{np.min(obs_zenith):.2f} deg < zenith angle < {np.max(obs_zenith):.2f} deg")
+print(f"{np.min(obs_muoneff):.2f} < muon efficiency < {np.max(obs_muoneff):.2f}")
+
+
+######################################################################
+# Manual grouping of observations
+# -------------------------------
+#
+# Here we can plot the zenith angle vs muon efficiency of the observations.
+# We decide to group the observations according to their zenith angle.
+# This is done manually as per a user defined cut, in this case we take the
+# median value of the zenith angles to define each observation group. These
+# are shown visually below.
+
+# This type of grouping can be utilised according to different parameters i.e.
+# zenith angle, muon efficiency, offset angle. The quantity chosen can therefore
+# be adjusted according to each specific science case.
+#
+
+fix, ax = plt.subplots(1, 1, figsize=(7, 5))
+obs_A = Observations([])
```

Comment:
maybe a more explicit name is better e.g. `observations_low_zenith`. Longer but clearer.